### PR TITLE
Implement low-touch tpgtools integration.

### PR DESCRIPTION
**Pull request by trodge**
Make it possible to implement new DCL resources in tpgtools without modifying mmv1.

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
Let me know if you need any help trouble shooting anything here. Happy to help and provide any and all assistance

**Review by ScottSuarez — CHANGES_REQUESTED** [review]
Take a look at dcl_client_creation

You'll likely need to source the new base path there.

https://github.com/GoogleCloudPlatform/magic-modules/blob/main/tpgtools/templates/provider_dcl_client_creation.go.tmpl#L45

**Comment by ScottSuarez** [comment]
Could you remove `serialization.go`? 

its been added to the `.gitignore` list

**Comment by trodge** [comment]
> 

Done

File: tpgtools/names.go [hunk 1]
```diff
@@ -125,6 +125,9 @@ func (b BasePathOverrideNameSnakeCase) ToTitle() RenderedString {
 	if strings.HasPrefix(string(b), "os") {
 		return RenderedString("OS" + title[2:])
 	}
+	if strings.HasPrefix(string(b), "gke") {
```
Comment (ScottSuarez) [inline tpgtools/names.go:128]: oh interesting.. nice find

File: mmv1/third_party/terraform/utils/config.go.erb [hunk 2]
```diff
@@ -200,30 +201,12 @@ type Config struct {
...
-	requestBatcherIam          *RequestBatcher
-
-	// start DCLBasePaths
-	// dataprocBasePath is implemented in mm
-	AssuredWorkloadsBasePath string
-	ClouddeployBasePath string
-	CloudResourceManagerBasePath string
+	// dcl
 	ContainerAwsBasePath string
 	ContainerAzureBasePath string
```
Comment (ScottSuarez) [inline mmv1/third_party/terraform/utils/config.go.erb:206]: Do you have context on why these entries need to be manually provided?

File: tpgtools/templates/provider_dcl_endpoints.go.tmpl [hunk 3]
```diff
@@ -47,24 +50,29 @@ var {{$pkg.BasePathIdentifier.ToTitle}}EndpointEntry = &schema.Schema{
...
 {{- range $index, $pkg := . }}
 {{- if $pkg.ShouldWriteProductBasePath }}
-// {{$pkg.BasePathIdentifier.ToTitle}}EndpointEntryKey:               {{$pkg.BasePathIdentifier.ToTitle}}EndpointEntry,
+	provider.Schema[{{$pkg.BasePathIdentifier.ToTitle}}EndpointEntryKey] = {{$pkg.BasePathIdentifier.ToTitle}}EndpointEntry
 {{- end}}
 {{- end}}
+}
 
-//Add new values to provider.go.erb - provider block read
+func providerDCLConfigure(ctx context.Context, d *schema.ResourceData, p *schema.Provider, config *Config) (interface{}, diag.Diagnostics) {
```
Comment (ScottSuarez) [inline tpgtools/templates/provider_dcl_endpoints.go.tmpl:69]:
```suggestion
func providerDCLConfigure(d *schema.ResourceData, config *Config) interface{} {
```
recommend not complicating the interface of this function to match the return parameters of the provider.go. If we have no need for diagnostics lets omit it from the return parameters. Same with the function parameters. If not needed, omit.

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

**Comment by ScottSuarez** [comment]
looks like tests were failing for azure and aws because the data source tries to [index into the basepath](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/data_sources/data_source_google_container_azure_versions.go#L56) `ContainerAzureBasePath`. This basepath is traditionally set as a default in [config.go](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/utils/config.go.erb#L270-L271) in the default base paths map. Then it is initialized here https://github.com/GoogleCloudPlatform/magic-modules/blob/07f36b0b334afd02d5bfa057f09ec5907e33b4aa/mmv1/third_party/terraform/utils/provider_handwritten_endpoint.go.erb#L172 .. 

unfortunately without this we have no way to know what the endpoint would be for the api since the dcl doesn't expose this.. Ideally the dcl could expose this so we could inherit it.

File: mmv1/third_party/terraform/utils/config.go.erb (hunk 2 above)
Comment (ScottSuarez) [reply mmv1/third_party/terraform/utils/config.go.erb:206]: looks like even though its dcl exclusive resource it needs a custom entry so the data source can have a default. It's a non-ideal behavior.

**Review by ScottSuarez — APPROVED** [review]
lgtm pending tests.

**Pull request merged**